stan/math/prim/mat/fun/multiply.hpp

```diff
@@ -52,24 +50,12 @@ multiply(T c, const Eigen::Matrix<double, R, C>& m) {
  * @throw std::domain_error if the number of columns of m1 does not match
  *   the number of rows of m2.
  */
-template <int R1, int C1, int R2, int C2>
-inline Eigen::Matrix<double, R1, C2> multiply(
-    const Eigen::Matrix<double, R1, C1>& m1,
-    const Eigen::Matrix<double, R2, C2>& m2) {
+template <int R1, int C1, int R2, int C2, typename T1, typename T2,
+          typename = enable_if_all_arithmetic<T1, T2>>
+inline Eigen::Matrix<return_type_t<T1, T2>, R1, C2> multiply(
+    const Eigen::Matrix<T1, R1, C1>& m1, const Eigen::Matrix<T2, R2, C2>& m2) {
   check_multiplicable("multiply", "m1", m1, "m2", m2);
-#ifdef STAN_OPENCL
-  if (m1.rows() * m1.cols() * m2.cols()
-      > opencl_context.tuning_opts().multiply_dim_prod_worth_transfer) {
-    matrix_cl<double> m1_cl(m1);
-    matrix_cl<double> m2_cl(m2);
-    matrix_cl<double> m3_cl = m1_cl * m2_cl;
-    return from_matrix_cl(m3_cl);
-  } else {
-    return m1 * m2;
-  }
-#else
```

Comment:
This should not have been removed. It was probably result of a bad merge or something. I will try to solve this by reapplying the changes of the original PR.

Reply:
Crap! Sorry this was def a bad merge on my part

Reply:
No problem Steve :) I am opening a new PR that just cherry-picks the previous merge of the OpenCL multiply PR. I think that will be less of a mess then reverting this PR and all that would follow.